Topshelf.Linux/LinuxConfigurationExtensions.cs

```diff
@@ -28,6 +29,7 @@ public static void UseLinuxIfAvailable(this HostConfigurator configurator)
 				// Needed to overcome mono-service style arguments.
 				configurator.ApplyCommandLine(MonoHelper.GetUnparsedCommandLine());
 				configurator.UseEnvironmentBuilder((cfg) => new LinuxHostEnvironmentBuilder(cfg));
+				configurator.UseHostBuilder((environment, settings) => new LinuxRunBuilder(environment, settings));
```

Comment:
From what I see, this will break those cases when the service is running under "mono-service" (as those are not intended to run under ConsoleRunHost.

AFAIK, you should modify LinuxHostEnvironment::CreateServiceHost(..) in order to instantiate a this new LinuxConsoleRunHost, instead of forcing a new builder to override the host everytime.